backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -1 +1,53 @@
-export default class TeamEventAttendanceDao {}
+import * as admin from 'firebase-admin';
+import { v4 as uuidv4 } from 'uuid';
+import { memberCollection, teamEventAttendanceCollection } from '../firebase';
+import { DBTeamEventAttendance } from '../types/DataTypes';
+
+export default class TeamEventAttendanceDao {
+  static async createTeamEventAttendance(
+    teamEventAttendance: TeamEventAttendance
+  ): Promise<TeamEventAttendance> {
+    const teamAttendanceRef: DBTeamEventAttendance = {
+      ...teamEventAttendance,
+      member: memberCollection.doc(teamEventAttendance.member.email),
+      ...(typeof teamEventAttendance.hoursAttended === 'number'
+        ? { hoursAttended: teamEventAttendance.hoursAttended }
+        : {}),
+      pending: true,
+      uuid: teamEventAttendance.uuid ? teamEventAttendance.uuid : uuidv4()
+    };
+    await teamEventAttendanceCollection.doc(teamAttendanceRef.uuid).set(teamAttendanceRef);
+    return teamEventAttendance;
+  }
+
+  static async updateTeamEventAttendance(
+    attendance: TeamEventAttendance
+  ): Promise<TeamEventAttendance> {
+    const teamAttendanceRef: DBTeamEventAttendance = {
+      ...attendance,
+      member: memberCollection.doc(attendance.member.email)
+    };
+
+    const docRef = await teamEventAttendanceCollection.doc(teamAttendanceRef.uuid);
+    const currentDoc = await docRef.get();
+    const currentAttendance = currentDoc.data() as DBTeamEventAttendance;
+
+    if (
+      teamAttendanceRef.hoursAttended === undefined &&
```

Comment:
Not sure I understand the logic here. Are we only replacing the `hoursAttended` field data if it's undefined? Wouldn't that be covered just with the line below (`await docRef.update(teamAttendanceRef);`)? Does it matter if the original field was undefined or not?

Reply:
This is probably related to what you did above with the `hoursAttended` field. Maybe there was some design issue discussed during work session that I'm not aware of that you could let me know about. I believe originally passing in undefined for hoursAttended field was fine, since we wouldn't use the field anyways if the event was not hour-based.

Reply:
Yeah, it's related. Ignore my comment for now while we wait for a response from @JacksonStaniec @henry-li-06.

Reply:
See above ^^ (`undefined` values will be enabled soon)